### PR TITLE
Deprecate helanal

### PR DESCRIPTION
**Pull request by lilyminium**
Changes made in this Pull Request:
 - Deprecate helanal from #2622 

Sorry for the delay in adding the helanal deprecation notice.

### CODE REVIEW

File: package/MDAnalysis/analysis/helanal.py [hunk 1]
```diff
@@ -116,9 +123,6 @@
 .. autofunction:: helanal_main
 
 """
-from __future__ import print_function, division, absolute_import
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/helanal.py:119]: If this is targetting 1.0.1 then these need to come back in right?

File: package/MDAnalysis/analysis/helanal.py [hunk 2]
```diff
@@ -131,6 +135,12 @@
 import logging
 logger = logging.getLogger("MDAnalysis.analysis.helanal")
 
+warnings.warn("This module is deprecated as of MDAnalysis version 1.0. "
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/helanal.py:138]: We'll probably want some test for this? (not sure how strict we are being with 1.0.1 tests, but if it's getting released we should try to get coverage down?)

Comment (IAlibay) [reply package/MDAnalysis/analysis/helanal.py:138]: Actually the tests are failing here, so we it'll need fixing anyways.

File: package/MDAnalysis/analysis/helanal.py [hunk 3]
```diff
@@ -243,7 +253,18 @@ def helanal_trajectory(universe, select="name CA",
 
     .. versionchanged:: 1.0.0
        Changed `selection` keyword to `select`
+
+    .. deprecated:: 1.0.1
+       helanal_trajectory is deprecated and will be removed in 2.0.0.
+       Please use helix_analysis.HELANAL instead (available in 2.0.0).
     """
+
+    warnings.warn("This function is deprecated as of MDAnalysis version 1.0. "
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/helanal.py:262]: I think we have this nice decorator for deprecations under lib.utils, you should be able to just do 

`@deprecate(release="1.0.1", remove="2.0", message="In 2.0 use the upcoming helix_analysis module.")`

for all these functions

File: package/MDAnalysis/analysis/helanal.py [hunk 4]
```diff
@@ -581,8 +602,18 @@ def helanal_main(pdbfile, select="name CA", ref_axis=None):
     .. versionchanged:: 1.0.0
        Changed `selection` keyword to `select`
 
+    .. deprecated:: 1.0.1
+       helanal_trajectory is deprecated and will be removed in 2.0.0.
+       Please use helix_analysis.HELANAL instead (available in 2.0.0).
+
     """
 
+    warnings.warn("This function is deprecated as of MDAnalysis version 1.0. "
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/helanal.py:611]: As above.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of changes, probably would be worth switching over to the deprecate decorator. Also we'll need to test warnings (or at least fix the test for warnings).

File: package/MDAnalysis/analysis/helanal.py (hunk 1 above)
Comment (orbeckst) [reply package/MDAnalysis/analysis/helanal.py:119]: yes

**Comment by orbeckst** [comment]
Just fyi, you can also make a PR against master, that's fine, too.

File: testsuite/MDAnalysisTests/analysis/test_helanal.py [hunk 5]
```diff
@@ -198,14 +198,13 @@ def test_warnings(tmpdir):
                 u, select="name CA", begin=-1, finish=99999
             )
 
-            assert len(record) == 2
+            assert len(record) == 3
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_helanal.py:201]: I guess that works as a really minimal test 👍

File: package/MDAnalysis/analysis/helanal.py [hunk 6]
```diff
@@ -117,20 +124,24 @@
...
 import logging
 logger = logging.getLogger("MDAnalysis.analysis.helanal")
 
+warnings.warn("This module is deprecated as of MDAnalysis version 1.0. "
+              "It will be removed in MDAnalysis version 2.0."
+              "Please use (available in 2.0.0) "
+              "MDAnalysis.analysis.helix_analysis instead.",
+              category=DeprecationWarning)
+
 def center(coordinates):
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/helanal.py:145]: These are all public facing functions, do we need to add deprecation warnings here too or do we just assume the warning on module importing will be sufficient?

**Review by IAlibay — APPROVED** [review]
lgtm, just a couple of comments that might be worth considering.

@orbeckst since this is going into your 1.0.1 PR I'll let you do the merge?

File: package/MDAnalysis/analysis/helanal.py (hunk 6 above)
Comment (orbeckst) [reply package/MDAnalysis/analysis/helanal.py:145]: I think the module import warning is sufficient.

File: testsuite/MDAnalysisTests/analysis/test_helanal.py (hunk 5 above)
Comment (orbeckst) [reply testsuite/MDAnalysisTests/analysis/test_helanal.py:201]: Curious that DeprecationWarning counts as a UserWarning... but ok, if it works...

**Comment by orbeckst** [comment]
Thank you @lilyminium and thanks @IAlibay for the review!

**Pull request merged**